File: Lab3/Lab3.1/app/src/main/java/ru/mirea/lab3/MainActivity.java
package ru.mirea.lab3;

import android.app.Activity;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.SimpleAdapter;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.HashMap;
import java.util.Locale;

public class MainActivity extends AppCompatActivity {

    static DBHelper mydb;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mydb = new DBHelper(MainActivity.this);
    }

    public void showDatabaseActivity (View v)
    {
        Intent i = new Intent( MainActivity.this, DatabaseActivity.class);
        startActivity(i);
    }

public void create5students (View v)
{
    Date currentDate = new Date();
    DateFormat finalDateAndTime = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss", Locale.getDefault());
    mydb.student5roulette(finalDateAndTime.format(currentDate));
}
public void russianvodka (View v)
{
    Date currentDate = new Date();
    DateFormat finalDateAndTime = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss", Locale.getDefault());
    mydb.ivanovtransformer(finalDateAndTime.format(currentDate));
}
public void addrandomrow (View v)
{
    Date currentDate = new Date();
    DateFormat finalDateAndTime = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss", Locale.getDefault());
    mydb.addrandomstudent(finalDateAndTime.format(currentDate));
}
}

File: Lab0/app/src/main/java/ru/mirea/lab0/MainActivity.kt
package ru.mirea.lab0

import android.content.Intent
import android.net.Uri
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val Linkbutton = findViewById<Button>(button2.id)
        val Exitbutton = findViewById<Button>(MAINbutton.id)

        Linkbutton.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                startActivity(Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/Destinyr4zr/DPA_labs_2019")))}
        })
        Exitbutton.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                finish()}
        })
}
}
